geometry_definitions.py

```diff
@@ -0,0 +1,160 @@
+                                                    fuel_assembly_lattice_cell])
+
+    return fuel_assembly_universe
+
+
+def tie_tube(hydrogen_inner, hydrogen_outer, inconel, ZrH, ZrC):
+    #THIS IS A GUESS BASED ON THE FUEL ELEMENT
+    assembly_cladding_thickness = 0.005
+    
+    inner_hydrogen_outer_radius = 0.20955
+    propellant_channel_pitch = 0.4089
+    inner_tie_tube_outer_radius = 0.26035
+    first_gap_outer_radius = 0.26670
+    moderator_outer_radius = 0.58420
+    second_gap_outer_radius = 0.67818
+    outer_tie_tube_outer_radius = 0.69850
+    third_gap_outer_radius = 0.70485
+    insulator_outer_radius = 0.80645
+    fourth_gap_outer_radius = 0.81280
+    flat_to_flat = 1.905
+    flat_to_flat_inner = flat_to_flat-2*assembly_cladding_thickness
+    assembly_edge_length = 0.5*flat_to_flat/np.cos(np.deg2rad(30))
+    inner_edge_length = 0.5*flat_to_flat_inner/np.cos(np.deg2rad(30))
+
+    # OpenMC Geometry
+    inner_hydrogen = openmc.ZCylinder(r=inner_hydrogen_outer_radius)
+    inner_tie_tube = openmc.ZCylinder(r=inner_tie_tube_outer_radius)
+    first_gap = openmc.ZCylinder(r=first_gap_outer_radius)
+    moderator_tube = openmc.ZCylinder(r=moderator_outer_radius)
+    second_gap = openmc.ZCylinder(r=second_gap_outer_radius)
+    outer_tie_tube = openmc.ZCylinder(r=outer_tie_tube_outer_radius)
+    third_gap = openmc.ZCylinder(r=third_gap_outer_radius)
+    insulator = openmc.ZCylinder(r=insulator_outer_radius)
+    fourth_gap = openmc.ZCylinder(r=fourth_gap_outer_radius)
+    tie_tube_assembly = openmc.model.HexagonalPrism(orientation='x',
+                                                    edge_length=inner_edge_length)
+    tie_tube_assembly_cladding = openmc.model.HexagonalPrism(orientation='x',
+                                                             edge_length=assembly_edge_length)
+
+    # OpenMC Cells and Universes
+    inner_hydrogen_cell = openmc.Cell(region=-inner_hydrogen, fill=hydrogen_inner)
+    inner_tie_tube_cell = openmc.Cell(
+        region=+inner_hydrogen & - inner_tie_tube, fill=inconel)
+    first_gap_cell = openmc.Cell(region=+inner_tie_tube & -first_gap)
+    moderator_tube_cell = openmc.Cell(region=+first_gap & -moderator_tube,fill=ZrH)
+    outer_hydrogen_cell = openmc.Cell(region=+moderator_tube & -second_gap, fill=hydrogen_outer)
+    outer_tie_tube_cell = openmc.Cell(region=+second_gap & -outer_tie_tube, fill=inconel)
+    third_gap_cell = openmc.Cell(region=+outer_tie_tube & -third_gap)
+    insulator_cell = openmc.Cell(region=+outer_tie_tube & -insulator,fill = ZrC)
+    fourth_gap_cell = openmc.Cell(region=+insulator & -fourth_gap)
+
+    # the fill on these two are also educated guesses, print statements are to ensure we fix this
+    tie_tube_assembly_cell = openmc.Cell(region=-tie_tube_assembly & +fourth_gap, fill=ZrH)
+    tie_tube_assembly_cladding_cell = openmc.Cell(
+        region=+tie_tube_assembly & -tie_tube_assembly_cladding, fill=ZrC)
```

Comment:
ZrC is correct here, and it should be 100% TD (which I say to differentiate it from the earlier comment about 50% TD ZrC)